graphql/schema/gqlschema.go

```diff
@@ -812,6 +839,13 @@ func addFilterType(schema *ast.Schema, defn *ast.Definition) {
 		}
 	}
 
+	// Has filter makes sense only if there is atleast one non ID field in the defn
+	if len(getNonIDFields(schema, defn)) > 0 {
```

Comment:
Actually sorry we should use `getFieldsWithoutIDType` here because that gives all fields except field having type `ID`. So it also gives fields with `@id` directive which is what we want because user can give the `Has` filter on a field with `@id` directive as well.